models/stics/R/model2netcdf.STICS.R

```diff
@@ -0,0 +1,89 @@
+#-------------------------------------------------------------------------------
+# Copyright (c) 2012 University of Illinois, NCSA.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the 
+# University of Illinois/NCSA Open Source License
+# which accompanies this distribution, and is available at
+# http://opensource.ncsa.illinois.edu/license.html
+#-------------------------------------------------------------------------------
+
+##-------------------------------------------------------------------------------------------------#
+##' Convert STICS output into the NACP Intercomparison format (ALMA using netCDF)
+##' 
+##' @name model2netcdf.STICS
+##' @title Code to convert STICS' output into netCDF format
+##'
+##' @param outdir Location of model output
+##' @param sitelat Latitude of the site
+##' @param sitelon Longitude of the site
+##' @param start_date Start time of the simulation
+##' @param end_date End time of the simulation
+##' @export
+##'
+##' @author Istem Fer
+model2netcdf.STICS <- function(outdir, sitelat, sitelon, start_date, end_date, overwrite = FALSE) {
+
+
+  ### Read in model output in STICS format
+  out_files <- list.files(outdir)
+  
+  stics_out_file <- file.path(outdir, out_files[grepl("mod_s.*", out_files)])
+  stics_output   <- read.table(stics_out_file, header = T, sep = ";")
+  output_names   <- colnames(stics_output)
+  
+  simulation_years <- unique(stics_output$ian)
+  
+  # get all years that we want data from
+  year_seq <- seq(lubridate::year(start_date), lubridate::year(end_date))
+  
+  # check that specified years and output years match: turning this off for now
+  # if (!all(year_seq %in% simulation_years)) {
+  #  PEcAn.logger::logger.severe("Years selected for model run and STICS output years do not match ")
+  #}
+  
+  # determine time step?
+  
+  for (y in simulation_years) {
+    
+    if (file.exists(file.path(outdir, paste(y, "nc", sep = "."))) & overwrite == FALSE) {
+      next
+    }
+    
+    thisyear <- stics_output[ , output_names == "ian"] == y
+    
+    outlist <- list()
+    outlist[[1]] <- stics_output[thisyear, which(output_names == "lai.n.")]  # LAI in (m2 m-2)
```

Comment:
Is there a reason to do `which(output_names == "lai.n.")` over just passing the column name `lai.n.`?

Reply:
just old code recycling